frappe/model/base_document.py

```diff
@@ -212,16 +212,23 @@ def update_if_missing(self, d):
 	def get_db_value(self, key):
 		return frappe.db.get_value(self.doctype, self.name, key)
 
-	def get(self, key, filters=None, limit=None, default=None):
+	def get(self, key, default=None, limit=None, filters=None):
 		if isinstance(key, dict):
 			return _filter(self.get_all_children(), key, limit=limit)
 
-		if filters:
-			if isinstance(filters, dict):
-				return _filter(self.__dict__.get(key, []), filters, limit=limit)
+		if filters is not None:
+			return _filter(self.__dict__.get(key, []), filters, limit=limit)
 
-			# perhaps you wanted to set a default instead
-			default = filters
+		if default and isinstance(default, dict):
+			import inspect
+
+			frm = inspect.stack()[1]
+			file = "/".join(frm.filename.split("/")[6:])
+			code = "".join(frm.code_context).strip()
+			frappe.log(f"Deprecated: please use doc.get filters keyword arg: {file}[{frm.lineno}]: {code}")
```

Comment:
Use `deprecation_warning` from `deprecations.py` instead of using inspect.

Reply:
Done